tests/suites/test_suite_cipher.aes.data

```diff
@@ -1098,6 +1098,626 @@ AES-256-ECB Decrypt NIST KAT #12
+
+AES-256-CBC crypt Decrypt NIST KAT #5
+depends_on:MBEDTLS_AES_C:MBEDTLS_CIPHER_MODE_CBC
+test_vec_crypt:MBEDTLS_CIPHER_AES_256_CBC:MBEDTLS_DECRYPT:"ffe0000000000000000000000000000000000000000000000000000000000000":"00000000000000000000000000000000":"d1ccb9b1337002cbac42c520b5d67722":"00000000000000000000000000000000":0
+
+AES-256-CBC crypt Decrypt NIST KAT #6
+depends_on:MBEDTLS_AES_C:MBEDTLS_CIPHER_MODE_CBC
+test_vec_crypt:MBEDTLS_CIPHER_AES_256_CBC:MBEDTLS_DECRYPT:"ffffc00000000000000000000000000000000000000000000000000000000000":"00000000000000000000000000000000":"27ef2495dabf323885aab39c80f18d8b":"00000000000000000000000000000000":0
+
+AES-256-CBC crypt Decrypt NIST KAT #7
+depends_on:MBEDTLS_AES_C:MBEDTLS_CIPHER_MODE_CBC
+test_vec_crypt:MBEDTLS_CIPHER_AES_256_CBC:MBEDTLS_DECRYPT:"ffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffe000":"00000000000000000000000000000000":"50e6d3c9b6698a7cd276f96b1473f35a":"00000000000000000000000000000000":0
+
+AES-256-CBC crypt Decrypt NIST KAT #8
+depends_on:MBEDTLS_AES_C:MBEDTLS_CIPHER_MODE_CBC
+test_vec_crypt:MBEDTLS_CIPHER_AES_256_CBC:MBEDTLS_DECRYPT:"c47b0294dbbbee0fec4757f22ffeee3587ca4730c3d33b691df38bab076bc558":"00000000000000000000000000000000":"46f2fb342d6f0ab477476fc501242c5f":"00000000000000000000000000000000":0
+
+AES-256-CBC crypt Decrypt NIST KAT #9
+depends_on:MBEDTLS_AES_C:MBEDTLS_CIPHER_MODE_CBC
+test_vec_crypt:MBEDTLS_CIPHER_AES_256_CBC:MBEDTLS_DECRYPT:"28d46cffa158533194214a91e712fc2b45b518076675affd910edeca5f41ac64":"00000000000000000000000000000000":"4bf3b0a69aeb6657794f2901b1440ad4":"00000000000000000000000000000000":0
+
+AES-256-CBC crypt Decrypt NIST KAT #10
+depends_on:MBEDTLS_AES_C:MBEDTLS_CIPHER_MODE_CBC
+test_vec_crypt:MBEDTLS_CIPHER_AES_256_CBC:MBEDTLS_DECRYPT:"c1cc358b449909a19436cfbb3f852ef8bcb5ed12ac7058325f56e6099aab1a1c":"00000000000000000000000000000000":"352065272169abf9856843927d0674fd":"00000000000000000000000000000000":0
+
+AES-256-CBC crypt Decrypt NIST KAT #11
+depends_on:MBEDTLS_AES_C:MBEDTLS_CIPHER_MODE_CBC
+test_vec_crypt:MBEDTLS_CIPHER_AES_256_CBC:MBEDTLS_DECRYPT:"0000000000000000000000000000000000000000000000000000000000000000":"00000000000000000000000000000000":"5c9d844ed46f9885085e5d6a4f94c7d7":"014730f80ac625fe84f026c60bfd547d":0
+
+AES-256-CBC crypt Decrypt NIST KAT #12
+depends_on:MBEDTLS_AES_C:MBEDTLS_CIPHER_MODE_CBC
+test_vec_crypt:MBEDTLS_CIPHER_AES_256_CBC:MBEDTLS_DECRYPT:"0000000000000000000000000000000000000000000000000000000000000000":"00000000000000000000000000000000":"a9ff75bd7cf6613d3731c77c3b6d0c04":"0b24af36193ce4665f2825d7b4749c98":0
+
+AES-256-CBC crypt Decrypt NIST KAT #13
+depends_on:MBEDTLS_AES_C:MBEDTLS_CIPHER_MODE_CBC
+test_vec_crypt:MBEDTLS_CIPHER_AES_256_CBC:MBEDTLS_DECRYPT:"0000000000000000000000000000000000000000000000000000000000000000":"00000000000000000000000000000000":"623a52fcea5d443e48d9181ab32c7421":"761c1fe41a18acf20d241650611d90f1":0
```

Comment:
Are all these tests necessary? It looks to me like these are the tests from the AES module that are being already run on that test suite. IMO, we should only aim to test the cipher glue code instead of the underlying crypto primitive in this case...

Reply:
Perhaps there are too many tests here.. I was trying to add some tests. But I think we should test the 'mbedtls_cipher_crypt' API in several modes and key sizes, since the API should be tested as well, I only added `ECB` and `CBC` as examples for mods that neeed and don't need IV

Reply:
Ok, I understand that, but please consider that the cipher layer is only an abstraction and all these tests are already running as part of other test suites. It would be nice to avoid the duplication as tests take time to execute and resources to maintain...